### PR TITLE
 update spacy verison to v2.1.x

### LINKED ISSUE #227: Not compatible with spaCy v2.1.0
**Describe the bug**

Fonduer is not compatible with spaCy v2.1.0

**To Reproduce**
1. Install spaCy v2.1.0
2. Run tests

**Expected behavior**

Be compatible with spaCy v2.1.0 and higher

**Error Logs/Screenshots**

```
jovyan@1a70f98cfba1:~/fonduer$ pip list | grep spacy
spacy                    2.1.2       
jovyan@1a70f98cfba1:~/fonduer$ pytest tests/parser/test_parser.py::test_parse_md_details
====================================================================================== test session starts ======================================================================================
platform linux -- Python 3.7.1, pytest-4.3.1, py-1.8.0, pluggy-0.9.0 -- /opt/conda/bin/python
cachedir: .pytest_cache
rootdir: /home/jovyan/fonduer/tests, inifile: pytest.ini
collected 1 item                                                                                                                                                                                

tests/parser/test_parser.py::test_parse_md_details FAILED                                                                                                                                 [100%]

=========================================================================================== FAILURES ============================================================================================
_____________________________________________________________________________________ test_parse_md_details _____________________________________________________________________________________

caplog = <_pytest.logging.LogCaptureFixture object at 0x7f4728ad9e48>

    def test_parse_md_details(caplog):
        """Test the parser with the md document."""
        caplog.set_level(logging.INFO)
        logger = logging.getLogger(__name__)
    
        docs_path = "tests/data/html_simple/md.html"
        pdf_path = "tests/data/pdf_simple/md.pdf"
    
        # Preprocessor for the Docs
        preprocessor = HTMLDocPreprocessor(docs_path)
        doc = next(preprocessor._parse_file(docs_path, "md"))
    
        # Check that doc has a name
        assert doc.name == "md"
    
        # Check that doc does not have any of these
        assert len(doc.figures) == 0
        assert len(doc.tables) == 0
        assert len(doc.cells) == 0
        assert len(doc.sentences) == 0
    
        # Create an Parser and parse the md document
        parser_udf = get_parser_udf(
            structural=True,
            tabular=True,
            lingual=True,
            visual=True,
            pdf_path=pdf_path,
            language="en",
        )
>       for _ in parser_udf.apply(doc):

tests/parser/test_parser.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/fonduer/parser/parser.py:241: in apply
    for _ in self.parse(document, text):
src/fonduer/parser/parser.py:810: in parse
    tokenized_sentences += [y for y in self._parse_node(node, state)]
src/fonduer/parser/parser.py:810: in <listcomp>
    tokenized_sentences += [y for y in self._parse_node(node, state)]
src/fonduer/parser/parser.py:755: in _parse_node
    yield from self._parse_paragraph(node, state)
src/fonduer/parser/parser.py:653: in _parse_paragraph
    yield from self._parse_sentence(paragraph, node, state)
src/fonduer/parser/parser.py:493: in _parse_sentence
    for parts in self.tokenize_and_split_sentences(document, text):
src/fonduer/parser/spacy_parser.py:275: in split_sentences
    sbd = self.model.create_pipe("sbd")  # add sentencizer
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <spacy.lang.en.English object at 0x7f472899ef60>, name = 'sbd', config = {}

    def create_pipe(self, name, config=dict()):
        """Create a pipeline component from a factory.
    
        name (unicode): Factory name to look up in `Language.factories`.
        config (dict): Configuration parameters to initialise component.
        RETURNS (callable): Pipeline component.
    
        DOCS: https://spacy.io/api/language#create_pipe
        """
        if name not in self.factories:
            if name == "sbd":
>               raise KeyError(Errors.E108.format(name=name))
E               KeyError: "[E108] As of spaCy v2.1, the pipe name `sbd` has been deprecated in favor of the pipe name `sentencizer`, which does the same thing. For example, use `nlp.create_pipeline('sentencizer')`"

/opt/conda/lib/python3.7/site-packages/spacy/language.py:251: KeyError
======================================================================================= warnings summary ========================================================================================
/opt/conda/lib/python3.7/site-packages/thinc/neural/train.py:7
  /opt/conda/lib/python3.7/site-packages/thinc/neural/train.py:7: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from .optimizers import Adam, linear_decay

/opt/conda/lib/python3.7/site-packages/thinc/check.py:4
/opt/conda/lib/python3.7/site-packages/thinc/check.py:4
/opt/conda/lib/python3.7/site-packages/thinc/check.py:4
  /opt/conda/lib/python3.7/site-packages/thinc/check.py:4: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Sequence, Sized, Iterable, Callable

/opt/conda/lib/python3.7/site-packages/jsonschema/compat.py:6
  /opt/conda/lib/python3.7/site-packages/jsonschema/compat.py:6: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import MutableMapping, Sequence  # noqa

parser/test_parser.py::test_parse_md_details
  /opt/conda/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 176 from C header, got 216 from PyObject
    return f(*args, **kwds)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================================================================= 1 failed, 6 warnings in 2.75 seconds ==============================================================================
```

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - PostgreSQL Version: 11.1-1.pgdg90+1
 - Poppler Utils Version: 0.62.0
 - Fonduer Version: 138e4be9a26ce7cd7086664dd9db22bcfb17b6b0

**Additional context**

### PR DESCRIPTION
**Pull request by SenWu**
Closes #227, #229.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
         "pyyaml>=4.2b1, <5.0",
         "scipy>=1.1.0, <2.0.0",
         "snorkel-metal>=0.4.1, <0.5.0",
-        "spacy>=2.0.12, <3.0.0",
+        "spacy>=2.1.2, <2.2.0",
```
Comment (HiromuHota) [inline setup.py]: Can this be `"spacy>=2.1.0, <2.2.0"`?

**Comment by lukehsiao** [comment]
Travis jobs are taking a long time because they are stalling during parsing.

```
----------------------------------- live log call -----------------------------------
[INFO] Using two cores.                                
[2019-03-25 14:35:24,418][INFO] fonduer.meta:129 - Connecting user:None to localhost:5432/cand_test
[INFO] Connecting user:None to localhost:5432/cand_test
[2019-03-25 14:35:24,652][INFO] fonduer.meta:156 - Initializing the storage schema
[INFO] Initializing the storage schema                                               
[2019-03-25 14:35:25,232][INFO] tests.candidates.test_candidates:215 - Parsing...
[INFO] Parsing...                                                    
[2019-03-25 14:35:25,452][INFO] fonduer.utils.udf:57 - Running UDF...
[INFO] Running UDF...                                              
  0%|                                                          | 0/1 [00:00<?, ?it/s]Process ParserUDF-2:
Traceback (most recent call last):                                  
  File "/usr/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()                                                                       
  File "/home/lwhsiao/repos/fonduer/src/fonduer/utils/udf.py", line 194, in run
    self.session.add_all(y for y in self.apply(doc, **self.apply_kwargs))        
  File "/home/lwhsiao/repos/fonduer/.venv/lib/python3.6/site-packages/sqlalchemy/orm/session.py", line 1784, in add_all
    for instance in instances:             
  File "/home/lwhsiao/repos/fonduer/src/fonduer/utils/udf.py", line 194, in <genexpr>
    self.session.add_all(y for y in self.apply(doc, **self.apply_kwargs))
  File "/home/lwhsiao/repos/fonduer/src/fonduer/parser/parser.py", line 241, in apply
    for _ in self.parse(document, text):
  File "/home/lwhsiao/repos/fonduer/src/fonduer/parser/parser.py", line 810, in parse
    tokenized_sentences += [y for y in self._parse_node(node, state)]
  File "/home/lwhsiao/repos/fonduer/src/fonduer/parser/parser.py", line 810, in <listcomp>
    tokenized_sentences += [y for y in self._parse_node(node, state)]
  File "/home/lwhsiao/repos/fonduer/src/fonduer/parser/parser.py", line 755, in _parse_node
    yield from self._parse_paragraph(node, state)                                    
  File "/home/lwhsiao/repos/fonduer/src/fonduer/parser/parser.py", line 653, in _parse_paragraph
    yield from self._parse_sentence(paragraph, node, state)
  File "/home/lwhsiao/repos/fonduer/src/fonduer/parser/parser.py", line 493, in _parse_sentence
    for parts in self.tokenize_and_split_sentences(document, text):         
  File "/home/lwhsiao/repos/fonduer/src/fonduer/parser/spacy_parser.py", line 310, in split_sentences
    for sent in doc.sents:                                                        
  File "doc.pyx", line 589, in sents                                                 
ValueError: [E030] Sentence boundaries unset. You can add the 'sentencizer' component to the pipeline with: nlp.add_pipe(nlp.create_pipe('sentencizer')) Alternatively, add the dependency parser, or set sentence boundaries by setting doc[i].is_sent_start.
```

This process has the `ValueError`, but the `test_cand_gen_cascading_delete` does not fail, it just hangs until the 50 minute timeout is reached.

For reference, these are the dependency versions I have when I see this error after a fresh `make dev`.
```
$ pip list
Package                  Version     Location                       
------------------------ ----------- -------------------------------
alabaster                0.7.11      
appdirs                  1.4.3       
aspy.yaml                1.1.1       
atomicwrites             1.2.1       
attrs                    18.2.0      
Babel                    2.6.0       
beautifulsoup4           4.6.3       
black                    18.9b0      
blis                     0.2.4       
boto3                    1.9.37      
botocore                 1.12.37     
bs4                      0.0.1       
cached-property          1.5.1       
certifi                  2018.8.24   
cfgv                     1.1.0       
chardet                  3.0.4       
Click                    7.0         
cycler                   0.10.0      
cymem                    2.0.2       
cytoolz                  0.9.0.1     
de-core-news-sm          2.0.0       
decorator                4.3.0       
dill                     0.2.8.2     
docutils                 0.14        
editdistance             0.5.2       
en-core-web-sm           2.1.0       
flake8                   3.5.0       
fonduer                  0.6.1       /home/lwhsiao/repos/fonduer/src
identify                 1.1.6       
idna                     2.7         
imagesize                1.1.0       
isort                    4.3.4       
Jinja2                   2.10        
jmespath                 0.9.3       
jsonschema               2.6.0       
kiwisolver               1.0.1       
lxml                     4.2.5       
MarkupSafe               1.0         
matplotlib               3.0.0       
mccabe                   0.6.1       
mecab-python3            0.7         
more-itertools           4.3.0       
msgpack                  0.5.6       
msgpack-numpy            0.4.4.1     
murmurhash               0.28.0      
mypy                     0.670       
mypy-extensions          0.4.1       
networkx                 2.2         
nltk                     3.3         
nodeenv                  1.3.2       
numpy                    1.15.2      
packaging                18.0        
pandas                   0.23.4      
pip                      19.0.2      
pkg-resources            0.0.0       
plac                     0.9.6       
pluggy                   0.7.1       
pre-commit               1.11.1      
preshed                  2.0.1       
protobuf                 3.7.0       
psycopg2-binary          2.7.5       
py                       1.6.0       
pyaws                    0.2.22      
pycodestyle              2.3.1       
pyflakes                 1.6.0       
Pygments                 2.2.0       
pyparsing                2.2.1       
pytest                   3.8.1       
python-dateutil          2.7.3       
pytz                     2018.5      
PyYAML                   4.2b4       
regex                    2017.4.5    
requests                 2.19.1      
s3transfer               0.1.13      
scikit-learn             0.20.0      
scipy                    1.1.0       
setuptools               40.4.3      
six                      1.11.0      
snorkel-metal            0.4.1       
snowballstemmer          1.2.1       
spacy                    2.1.3       
Sphinx                   1.8.1       
sphinx-rtd-theme         0.4.1       
sphinxcontrib-websupport 1.1.0       
SQLAlchemy               1.2.12      
srsly                    0.0.5       
style                    1.1.0       
tensorboardX             1.6         
thinc                    7.0.4       
toml                     0.9.6       
toolz                    0.9.0       
torch                    1.0.1.post2 
tqdm                     4.26.0      
treedlib                 0.1.1       
typed-ast                1.3.1       
ujson                    1.35        
urllib3                  1.23        
virtualenv               16.0.0      
Wand                     0.4.4       
wasabi                   0.2.0       
wheel                    0.31.1      
wrapt                    1.10.11     
```

**Comment by lukehsiao** [comment]
Tests appear to work when using space `2.1.2`, but fail with space `2.1.3`.

**Comment by HiromuHota** [comment]
> Tests appear to work when using space `2.1.2`, but fail with space `2.1.3`.

I confirmed this on my side.

File: .travis.yml [hunk 2]
```diff
@@ -99,7 +102,7 @@ before_script:
 - cd ..
 
 script:
-- travis_wait 50 coverage run --source=fonduer -m pytest tests
+- travis_wait 60 coverage run --source=fonduer -m pytest tests
```
Comment (lukehsiao) [inline .travis.yml]:
```suggestion
- travis_wait 50 coverage run --source=fonduer -m pytest tests
```

This should remain 50, as 50 min is the upper-limit on a Travis build anyways.

**Pull request merged**